### PR TITLE
Remove JavaScript and CSS type attributes for HTML5 validity

### LINKED ISSUE #1118: Allow CP to produce HTML5 compliant markup
## Expected behavior
CP should produce HTML5 compliant code if adding theme support for it using:
`add_theme_support( 'html5', array( 'comment-list', 'comment-form', 'search-form', 'gallery', 'caption' ) );`

## Current behavior
It does not produce HTML5 compliant markup for scripts and styles.

## Possible solution
Backport this change:
https://core.trac.wordpress.org/changeset/46164
It was introduced in WP 5.3 (adds support for script and style HTML5 compliant code in theme support declaration)

## Steps to reproduce (for bugs)
Validate any CP website in https://validator.w3.org/nu/
It will return errors for link tags using trailing slashes and script tags using the type attribute.

## Context
It should be possible to produce HTML5 compliant code with ClassicPress. Currently, it is xHTML compliant only.
While this is the "default" for both WP and CP, it has been a few years HTML5 is around, and is likely to be considered the standard.

## Notes
WordPress _still_ produces trailing slashes for the link style tag. This is wrong. It should not produce the closing trailing slash for _anything_, actually, if HTML5 complaint (`<br />` is another such case that WP still gets wrong)

Thus I think while doing the back port, we should also enhance it and make sure it _truly_ produces valid HTML5 when declared.

### PR DESCRIPTION
**Pull request by mattyrob**
Closes #1118

## Description
This is a back port of several upstream fixes:
https://core.trac.wordpress.org/changeset/43563
https://core.trac.wordpress.org/changeset/43564
https://core.trac.wordpress.org/changeset/46164
https://core.trac.wordpress.org/changeset/46165
https://core.trac.wordpress.org/changeset/46170
https://core.trac.wordpress.org/changeset/46171
https://core.trac.wordpress.org/changeset/46287

## Motivation and context
This is a second attempt at this enhancement, #524 was reverted in #876 recently due to errors seen with script concatenation.

The aim is to remove type attributes when adding style and javascript to a page in a theme that supports HTML5, Type attributes are not required by HTML5 and their inclusion creates warnings in HTML validators.

This PR includes a further 2 previous changes sets - 43563 is a simple documentation fix but 43564 improved the logic and made our version of `wp-includes/class.wp-styles.php` more consistent with the upstream code which improved subsequent backporting.

## How has this been tested?
Given the issues seen last time, the follwoing tests have been done:
Local PHPunit testing
Visual check of localhost performance with WP_DEBUG defined as true
Visual check of localhost performance with WP_DEBUG defined as false
Build code using `grunt build` and repeat checks above locally.

## Screenshots
N/A

## Types of changes
- Enhancement

### CODE REVIEW

File: tests/phpunit/tests/dependencies/scripts.php [hunk 1]
```diff
@@ -182,6 +184,23 @@ function test_wp_enqueue_script_filter_null_version() {
...
 	}
 
+	/**
+	 * @see https://core.trac.wordpress.org/ticket/42804
+	 */
+	function test_wp_enqueue_script_with_html5_support_does_not_contain_type_attribute() {
+		add_theme_support( 'html5', array( 'script' ) );
+
+		$GLOBALS['wp_scripts']                  = new WP_Scripts();
+		$GLOBALS['wp_scripts']->default_version = get_bloginfo( 'version' );
```
Comment (joyously) [inline tests/phpunit/tests/dependencies/scripts.php:194]: Not that this PR should be changing this, but shouldn't the default version be `classicpress_asset_version()`?

File: tests/phpunit/tests/dependencies/styles.php [hunk 2]
```diff
@@ -186,6 +188,23 @@ function test_wp_enqueue_style_filter_null_version() {
...
 	}
 
+	/**
+	 * @see https://core.trac.wordpress.org/ticket/42804
+	 */
+	function test_wp_enqueue_style_with_html5_support_does_not_contain_type_attribute() {
+		add_theme_support( 'html5', array( 'style' ) );
+
+		$GLOBALS['wp_styles']                  = new WP_Styles();
+		$GLOBALS['wp_styles']->default_version = get_bloginfo( 'version' );
```
Comment (joyously) [inline tests/phpunit/tests/dependencies/styles.php:198]: Shouldn't the default version be `classicpress_asset_version()`?

File: tests/phpunit/tests/dependencies/scripts.php (hunk 1 above)
Comment (mattyrob) [reply tests/phpunit/tests/dependencies/scripts.php]: It really doesn't matter in the unit tests as the version is defined twice for the test later, that said we should probably set a good example of how to use `classicpress_asset_version()`, PR update to follow.

**Comment by bahiirwa** [comment]
Closes #1118

**Pull request merged**